### PR TITLE
Created Markdown checker

### LINKED ISSUE #30: Add markdown link checker to avoid broken links
Example workflow from hazelcast-nodejs-client: https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/.github/workflows/markdown_link_checker.yml

### PR DESCRIPTION
**Pull request by RishiKumarRay**
fix #30

### CODE REVIEW

**Comment by srknzl** [comment]
One more thing, we can make the checker run when a PR is opened **and when it is merged to main**. You can achieve that by making `on:` part the following

```
on:
    push:
        branches:
            - main
    pull_request:
```

File: .github/workflows/markdown_link_checker.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 on: [pull_request]
```
Comment (srknzl) [inline .github/workflows/markdown_link_checker.yml:1]: you need to delete this line as well. There are two `on:` now

**Review by srknzl — CHANGES_REQUESTED** [review]
Thanks for the PR, almost done

**Comment by RishiKumarRay** [comment]
@srknzl please review now

File: .github/workflows/markdown_link_checker.yml (hunk 1 above)
Comment (srknzl) [reply .github/workflows/markdown_link_checker.yml:1]: @RishiKumarRay This is still not a valid workflow. This is what I wanted:


```
name: Check links for markdown files
on:
    push:
        branches:
            - main
    pull_request:
jobs:
  markdown-link-check:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: gaurav-nelson/github-action-markdown-link-check@v1
      with:
        use-quiet-mode: 'yes'
```

Comment (RishiKumarRay) [reply .github/workflows/markdown_link_checker.yml]: @srknzl check now

**Review by srknzl — APPROVED** [review]

**Comment by srknzl** [comment]
The link checker fails but I will fix the broken links in a separate PR. Thanks for the contribution.

**Comment by RishiKumarRay** [comment]
Thank you

**Pull request merged**